### PR TITLE
Backport 14264, 14324, 14583, 14426, 13687

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by PastaPastaPasta** [comment]
I believe all should be resolved here

File: test/functional/feature_config_args.py [hunk 1]
```diff
@@ -92,15 +92,18 @@ def run_test(self):
...
-        self.stop_node(0)
-        assert os.path.exists(os.path.join(new_data_dir, self.chain, 'blocks'))
-        if self.is_wallet_compiled():
-            assert os.path.exists(os.path.join(new_data_dir, 'regtest', 'wallets', 'w1'))
+        # Temporarily disabled, because this test would access the user's home dir (~/.bitcoin)
+        #self.start_node(0, ['-conf='+conf_file, '-wallet=w1'])
+        #self.stop_node(0)
+        #assert os.path.exists(os.path.join(new_data_dir, self.chain, 'blocks'))
+        #if self.is_wallet_compiled():
+        #    assert os.path.exists(os.path.join(new_data_dir, 'regtest', 'wallets', 'w1'))
```
Comment (UdjinM6) [inline test/functional/feature_config_args.py:106]: 13687:
```suggestion
        #    assert os.path.exists(os.path.join(new_data_dir, self.chain, 'wallets', 'w1'))
```

 _ideally_ this should be fixed in 14324 which should be backported _after_ 13687

File: test/functional/feature_notifications.py [hunk 2]
```diff
@@ -36,35 +35,36 @@ def setup_network(self):
...
-        # directory content should equal the generated transaction hashes
-        txids_rpc = list(map(lambda t: t['txid'], self.nodes[1].listtransactions("*", block_count)))
-        assert_equal(sorted(txids_rpc), sorted(os.listdir(self.walletnotify_dir)))
-        self.stop_node(1)
+            # directory content should equal the generated transaction hashes
+            txids_rpc = list(map(lambda t: t['txid'], self.nodes[1].listtransactions("*", block_count)))
+            assert_equal(sorted(txids_rpc), sorted(os.listdir(self.walletnotify_dir)))
+            self.stop_node(1)
         for tx_file in os.listdir(self.walletnotify_dir):
             os.remove(os.path.join(self.walletnotify_dir, tx_file))
```
Comment (UdjinM6) [inline test/functional/feature_notifications.py:56]: 14324: move these lines to the right (inside `if` block`)

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/feature_config_args.py (hunk 1 above)
Comment (PastaPastaPasta) [reply test/functional/feature_config_args.py:106]: But is that really important?

Comment (UdjinM6) [reply test/functional/feature_config_args.py:106]: would be cleaner/easier to review this way imo

File: test/functional/feature_config_args.py [hunk 3]
```diff
@@ -104,13 +101,18 @@ def run_test(self):
         # Temporarily disabled, because this test would access the user's home dir (~/.bitcoin)
         #self.start_node(0, ['-conf='+conf_file, '-wallet=w1'])
         #self.stop_node(0)
+        #assert os.path.exists(os.path.join(new_data_dir, self.chain, 'blocks'))
+        #if self.is_wallet_compiled():
         #assert os.path.exists(os.path.join(new_data_dir, self.chain, 'wallets', 'w1'))
 
-        # Ensure command line argument overrides datadir in conf
+        # Ensu#re command line argument overrides datadir in conf
```
Comment (UdjinM6) [inline test/functional/feature_config_args.py:108]: 14324: `Ensu#re` ->`Ensure`

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/feature_config_args.py (hunk 3 above)
Comment (PastaPastaPasta) [reply test/functional/feature_config_args.py]: done. Weird

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**